File: 128-longest-consecutive-sequence/longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        int n=nums.size();
        unordered_set<int> s;
        int maxLen=0;
        for(int i=0;i<n;i++) s.insert(nums[i]);

        for(auto it: s)
        {
            if(s.find(it-1)==s.end())
            {
                int x=it;
                int len=0;
                while(s.find(x)!=s.end())
                {
                    len++;
                    x++;
                }
                maxLen=max(maxLen,len);
            }
        }
        return maxLen;


        // if(n==0) return 0;
        // sort(nums.begin(),nums.end());
        // int count=1,maxLen=1;
        // for(int i=1;i<n;i++)
        // {
        //     if(nums[i]-nums[i-1]==0) continue;
        //     if(nums[i]-nums[i-1]==1) count++;
        //     else count=1;
        //     maxLen=max(maxLen,count);
        // }
        // return maxLen;
    }
};

File: 128-longest-consecutive-sequence/longest-consecutive-sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        HashSet<Integer> st = new HashSet<>();
        for(int num : nums) {
            st.add(num);
        }

        int maxLength = 0;
        for(int num : st) {
            if(st.contains(num-1)) continue;
            int count = 1;
            int val = num + 1;
            while(st.contains(val)) {
                count++;
                val++;
            }
            maxLength = Math.max(maxLength,count);
        }
        return maxLength;
    }
}